extensions/HyperVExtension/src/HyperVExtension/Models/HyperVVirtualMachine.cs

```diff
@@ -806,6 +810,48 @@ public SDK.ApplyConfigurationResult ApplyConfiguration(ApplyConfigurationOperati
         }
     }
 
+    private void ThrowIfApplyConfigurationIsNotSupported(GuestKvpSession guestSession)
+    {
+        Dictionary<string, string>? osVersionInfo = null;
+        var isApplyConfigurationSupported = false;
+
+        try
+        {
+            osVersionInfo = GetOsVersionInfo(guestSession);
+            var osPlatformId = int.Parse(osVersionInfo[HyperVStrings.OSPlatformId], CultureInfo.InvariantCulture);
+            if (osPlatformId == Constants.GuestPlatformIdWindows)

```

Comment:
Would we ever be worried about Case sensitivity? This has been around for so long where they probably will never change these, but figured I'd ask

Reply:
Not sure. I'll change it to be on the safe side.

Reply:
Actually, the dictionary returned from GetOsVersionInfo->ReadGuestProperties is StringComparer.OrdinalIgnoreCase.
I'll add a comment.